xgcm/test/test_metrics.py

```diff
@@ -44,3 +70,275 @@ def test_metrics_2d_grid():
+                "dz_w",
+                "area_ne",
+                "area_n",
+                "area_e",
+                "area_t",
+                "volume_t",
+            ],
+            [
+                dx_ne,
+                dx_n,
+                dx_e,
+                dx_t,
+                dy_ne,
+                dy_n,
+                dy_e,
+                dy_t,
+                dz_t,
+                dz_w,
+                area_ne,
+                area_n,
+                area_e,
+                area_t,
+                volume_t,
+            ],
+        ):
+            obj.coords[name] = data
+            obj.coords[name].attrs["tracked_name"] = name
+        # add xgcm attrs
+        for ii in ["xu", "xt"]:
+            obj[ii].attrs["axis"] = "X"
+        for ii in ["yu", "yt"]:
+            obj[ii].attrs["axis"] = "Y"
+        for ii in ["zt", "zw"]:
+            obj[ii].attrs["axis"] = "Z"
+        for ii in ["time"]:
+            obj[ii].attrs["axis"] = "T"
+        for ii in ["xu", "yu", "zw"]:
+            obj[ii].attrs["c_grid_axis_shift"] = 0.5
+        return obj
+
+    coords = {
+        "X": {"center": "xt", "right": "xu"},
+        "Y": {"center": "yt", "right": "yu"},
+        "Z": {"center": "zt", "right": "zw"},
+    }
+
+    metrics = {
+        ("X",): ["dx_t", "dx_n", "dx_e", "dx_ne"],
+        ("Y",): ["dy_t", "dy_n", "dy_e", "dy_ne"],
+        ("Z",): ["dz_t", "dz_w"],
+        ("X", "Y"): ["area_t", "area_n", "area_e", "area_ne"],
+        ("X", "Y", "Z"): ["volume_t"],
+    }
+    # Used previously for some fail test. Retain for now.
+    # coords_outer = {
+    #     "X": {"center": "xt", "outer": "xu"},
+    #     "Y": {"center": "yt", "outer": "yu"},
+    #     "Z": {"center": "zt", "outer": "zw"},
+    # }
+
+    # combine to different grid configurations (B and C grid)
+    ds_b = _add_metrics(
+        xr.Dataset(
+            {"u": u_b, "v": v_b, "wt": wt, "tracer": tr, "timeseries": timeseries}
+        )
+    )
+    ds_c = _add_metrics(
+        xr.Dataset(
+            {"u": u_c, "v": v_c, "wt": wt, "tracer": tr, "timeseries": timeseries}
+        )
+    )
+
+    # Some fail tests used in the weighted mean (retain for later)
+    # ds_fail = _add_metrics(
+    #     xr.Dataset(
+    #         {"u": u_b, "v": v_c, "tracer": tr, "timeseries": timeseries}
+    #     )
+    # )
+    # ds_fail2 = _add_metrics(
+    #     xr.Dataset(
+    #         {"u": u_b, "v": v_c, "tracer": tr, "timeseries": timeseries}
+    #     )
+    # )
+
+    return {
+        "B": ds_b,
+        "C": ds_c,
+        # "fail_gridtype": ds_fail,
+        # "fail_dimtype": ds_fail2,
+        "coords": coords,
+        "metrics": metrics,
+        # "fail_coords": coords_outer,
+    }
+
+
+@pytest.mark.parametrize(
+    "key, metric_vars",
+    [
+        # (("X"), ["dx_t"]), # this fails, can we raise a more useful error?
+        (("X",), ["dx_t"]),
+        (("X",), []),  # Should this pass?
+        (("X",), "dx_t"),
```

Comment:
This is the only test that is failing. I require the metrics to be a list. What do you think about that?

Reply:
For the sake of useability I would prefer to convert any string input to a list internally, but if a proper error is raised I could live without it. Great that the other all pass.

Reply:
What do you suggest is the best way to do this?

Reply:
I made a suggestion in the definition of `_assign_metrics`. 

Not sure if this is good practice, but the more I think about it, it would be a nice thing to have. I am again thinking of a very simple use case of e.g. an obervational dataset that has only the tracer cell dimensions. I think `Grid(..., metrics={'X':'dx, 'Y':'dy'})` looks a lot cleaner than `Grid(..., metrics={'X':['dx], 'Y':['dy']})`